runelite-client/src/main/java/net/runelite/client/plugins/itemstats/ItemStatChangesMethods.java

```diff
@@ -0,0 +1,336 @@
+			assert prev == null : "Item already added";
+		}
+	}
+
+	/**
+	 * StatChangeCalculator calculates a {@link ItemStatChange} for the item it is assigned to.
+	 * This is responsible for calculating or delegating individual {@link ItemStatChange.StatChange}s
+	 * and for calculating the overall positivity of the change.
+	 *
+	 * @see Combo
+	 * @see SuperRestore
+	 */
+	@FunctionalInterface
+	interface StatChangeCalculator
+	{
+		ItemStatChange calculate();
+	}
+
+	/**
+	 * Combo takes an list of {@link SingleStatChangeCalculator}s and combines them into
+	 * a single StatChangeCalculator.
+	 *
+	 * <code>numPrimaries</code> may be passed as the first parameter to limit the number
+	 * of "Primary" stats to be used for calculating the {@link ItemStatChange}'s overall
+	 * positivity
+	 */
+	static class Combo implements StatChangeCalculator
+	{
+		private final SingleStatChangeCalculator[] calcs;
+		private final int numPrimaries;
+
+		Combo(SingleStatChangeCalculator... calcs)
+		{
+			this(1, calcs);
+		}
+
+		Combo(int numPrimaries, SingleStatChangeCalculator... calcs)
+		{
+			assert numPrimaries <= calcs.length;
+			this.numPrimaries = numPrimaries;
+			this.calcs = calcs;
+		}
+
+		public ItemStatChange calculate()
+		{
+			ItemStatChange out = new ItemStatChange(calcs.length);
+			for (int i = 0; i < calcs.length; i++)
+			{
+				out.statChanges[i] = calcs[i].calculate();
+			}
+			int positivity = Positivity.NO_CHANGE;
+			for (int i = 0; i < numPrimaries; i++)
+			{
+				if (positivity < out.statChanges[i].positivity)
+				{
+					positivity = out.statChanges[i].positivity;
+				}
+			}
+			out.positivity = positivity;
+			return out;
+		}
+	}
+
+	/**
+	 * SingleStatChangeCalculator is responsible for calculating a single stat change
+	 */
+	@FunctionalInterface
+	interface SingleStatChangeCalculator
+	{
+		ItemStatChange.StatChange calculate();
+	}
+
+	/**
+	 * DeltaCalculator calculates the change for a stat given it's maximum.
+	 * It is a convenience interface for {@link SimpleStatCalc}. It should not
+	 * attempt to limit itself based on the <code>max</code> passed to it
+	 */
+	@FunctionalInterface
+	interface DeltaCalculator
+	{
+		int calculateDelta(int max);
+	}
+
+	/**
+	 * BOOST is set if the stat's cap should be increased by the calculated delta.
+	 * This behaves like most of RuneScape's "Temporary skill boosts"
+	 *
+	 * @see SimpleStatCalc
+	 */
+	static final int BOOST = 1;
+
+	/**
+	 * A {@link SingleStatChangeCalculator} for simple stat changes.
+	 *
+	 * @See add
+	 * @See perc
+	 */
+	SingleStatChangeCalculator simple(Stat stat, DeltaCalculator calc, int flags)
+	{
+		return new SimpleStatCalc(stat, calc, flags);
+	}
+
+	SingleStatChangeCalculator simple(Stat stat, DeltaCalculator calc)
+	{
+		return new SimpleStatCalc(stat, calc, 0);
+	}
+
+	@AllArgsConstructor
+	private static class SimpleStatCalc implements SingleStatChangeCalculator
+	{
+		protected Stat stat;
```

Comment:
Private, and use @Getter(accessLevel = Protected) to provide the fields to children.